Restore `setCreator` API dropped by accident in a minor release

lekoala opened the pull request:
this will also require Properties to not be readonly

fixes https://github.com/openspout/openspout/issues/286

lekoala committed:
restore setCreator

this will also require Properties to not be readonly

lekoala committed:
remove readonly

Slamdunk commented on src/Writer/XLSX/Properties.php:
```diff
@@ -4,7 +4,7 @@
 namespace OpenSpout\Writer\XLSX;
 
-final readonly class Properties
+final class Properties
```
This is not needed, just overwrite the object with a new one

Slamdunk commented on src/Writer/XLSX/Writer.php:
```diff
@@ -35,6 +35,11 @@ public function getOptions(): Options
         return $this->options;
     }
 
+    public function setCreator(string $creator): void
```
I guess we also need to restore this in the `WriterInterface`

lekoala committed:
replace properties

lekoala committed:
keep readonly

lekoala committed:
restore interface method

lekoala committed:
add abstract function

lekoala committed:
add method

lekoala commented:
@Slamdunk kept it readonly (it's just a bit verbose imho), added back method to interface. instead of implementing in abstract, i kept an abstract method and added a noop function to the csv writer

Slamdunk commented on src/Writer/AbstractWriter.php:
```diff
@@ -137,6 +137,8 @@ final public function close(): void
         $this->isWriterOpened = false;
     }
 
+    abstract public function setCreator(string $creator): void;
```
This is already enforced by the interface and can be dropped here

Slamdunk commented on src/Writer/CSV/Writer.php:
```diff
@@ -28,6 +28,11 @@ public function getOptions(): Options
+    
+    public function setCreator(string $creator): void
+    {
+        // noop
```
May it would be better to raise an Exception message like `Method unsopported for CSV documents`

lekoala committed:
throw exception

lekoala committed:
remove abstract

lekoala commented:
@Slamdunk i've made the requested changes

i'm not planning on updating this pr further than that, it feels like you would spend as much time writing the comment as doing the change yourself which seem counterproductive. my proposal would be to merge as is and make any last changes as you see fit.

have a great day,

Slamdunk commented:
> it feels like you would spend as much time writing the comment as doing the change yourself which seem counterproductive

It depends on the definition of counterproductive.
I review PR on the open source projects I maintain to share my knowledge so that the knowledge itself can improve for everyone from anyone has better ideas and suggestions.

If you feel annoyed by my feedbacks, imagine how I feel having to deal with hundreds of people like you after I spent my free time in the pursuit of reciprocal help.

Slamdunk committed:
CS Fix
